src/app/shared/state/entry-wizard.service.ts

```diff
@@ -0,0 +1,141 @@
+import { Injectable } from '@angular/core';
+import { EntryWizardStore } from './entry-wizard.store';
+import { DefaultService, BioWorkflow } from '../openapi';
+import { AlertService } from '../alert/state/alert.service';
+import { Repository } from '../openapi/model/repository';
+import { HttpErrorResponse } from '@angular/common/http';
+import { finalize } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EntryWizardService {
+  constructor(private entryWizardStore: EntryWizardStore, private defaultService: DefaultService, private alertService: AlertService) {}
+
+  /**
+   * Updates the list of git registries
+   */
+  updateGitRegistries() {
+    this.entryWizardStore.setLoading(true);
+    this.defaultService
+      .getUserRegistries()
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (registries: Array<string>) => {
+          this.entryWizardStore.update({ gitRegistries: registries });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of organizations based on the given git registry
+   * @param registry git registry
+   */
+  updateGitOrganizations(registry: string) {
+    this.entryWizardStore.setLoading(true);
+    this.entryWizardStore.update({ gitOrganizations: undefined, gitRepositories: undefined });
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .getUserOrganizations(registryEnum)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (organizations: Array<string>) => {
+          this.entryWizardStore.update({ gitOrganizations: organizations });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of repositories based on the given git registry and organization
```

Comment:
I find the word "update" in this comment and the function names a little confusing in comparison to "addWorkflow" -- this is only updating the Akita store, but addWorkfow is updating the backend, if I'm understanding correctly.

Maybe not a big deal, but when I saw `updateGitRepositories`, I was expecting to see something like `this.defaultService.update...`.